hikari/impl/bot.py

```diff
@@ -576,7 +576,15 @@ def run(
         if shard_ids is not None and shard_count is None:
             raise TypeError("'shard_ids' must be passed with 'shard_count'")
 
-        loop = asyncio.get_event_loop()
+        # get_event_loop will error under oddly specific cases such as if set_event_loop has been called before even
+        # if it was just called with None or if it's called on a thread which isn't the main Thread.
+        try:
+            loop = asyncio.get_event_loop()
+
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
```

Comment:
Just to continue the discussion we were having before, wouldn't it be better to have run actually take control of the event loop?

The way asyncio.run manages it is making sure there is no running event loop and then creating a new one. They also set the running event loop to `None`.

*This could be done in the form of an option passed into `run` that toggles this (manage_loop?), and then we could deprecate and eventually remove `close_loop` in favor for this option*

Reply:
Maybe this could be moved to a separate issue?